cli/src/main/java/com/devonfw/tools/ide/tool/intellij/IntellijPluginManager.java

```diff
@@ -0,0 +1,140 @@
+package com.devonfw.tools.ide.tool.intellij;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.io.FileAccess;
+import com.devonfw.tools.ide.step.Step;
+import com.devonfw.tools.ide.tool.ide.PluginDescriptor;
+
+/**
+ * Manager class to install plugins for the {@link Intellij commandlet}.
+ */
+public class IntellijPluginManager {
+
+  private static final String BUILD_FILE = "build.txt";
+  private final IdeContext context;
+  private final Intellij commandlet;
+
+  /**
+   * The constructor
+   *
+   * @param context the {@link IdeContext}.
+   * @param commandlet the {@link Intellij commandlet}.
+   */
+  public IntellijPluginManager(IdeContext context, Intellij commandlet) {
+    this.context = context;
+    this.commandlet = commandlet;
+  }
+
+  /**
+   * Installs a plugin.
+   *
+   * @param plugin the {@link PluginDescriptor}#.
+   */
+  public void installPlugin(PluginDescriptor plugin) {
+    String pluginId = plugin.getId();
+    String downloadUrl = plugin.getUrl();
+
+    Path tmpDir = null;
+
+    Step step = this.context.newStep("Install plugin: " + pluginId);
+    try {
+      String buildVersion = readBuildVersion();
+
+      Path installationPath = commandlet.getPluginsInstallationPath();
+      ensureInstallationPathExists(installationPath);
+
+      if (downloadUrl == null || downloadUrl.isEmpty()) {
+        downloadUrl = String.format("https://plugins.jetbrains.com/pluginManager?action=download&id=%s&build=%s", pluginId, buildVersion);
+      }
+
+      FileAccess fileAccess = context.getFileAccess();
+      tmpDir = fileAccess.createTempDir(pluginId);
+
+      Path downloadedFile = downloadPlugin(fileAccess, downloadUrl, tmpDir, buildVersion, pluginId);
+      extractDownloadedPlugin(fileAccess, downloadedFile, pluginId);
+
+      step.success();
+
+    } catch (IOException e) {
+      step.error(e);
+      throw new IllegalStateException("Failed to process installation of plugin: " + pluginId, e);
+    } finally {
+      if (tmpDir != null) {
+        context.getFileAccess().delete(tmpDir);
+      }
+      step.close();
+    }
+  }
+
+  private String readBuildVersion() throws IOException {
+    Path buildFile = commandlet.getToolPath().resolve(BUILD_FILE);
+    if (context.getSystemInfo().isMac()) {
+      buildFile = context.getSoftwareRepositoryPath().resolve("default").resolve("intellij/intellij").resolve(commandlet.getInstalledVersion().toString())
+          .resolve("IntelliJ IDEA" + commandlet.generateMacEditionString() + ".app").resolve("Contents/Resources").resolve(BUILD_FILE);
```

Comment:
This can probably be optimized, the resolve of the path to the app looks a bit too specific to me and maybe other methods e.g. `MacOsHelper` can be useful here too.